### PR TITLE
Add Agent Manager for multi-agent orchestration

### PR DESCRIPTION
**Pull request by marius-kilocode**
Adds a new Agent Manager panel for running multiple autonomous agent sessions in parallel.

**Features:**
- Adds an Agent Manager button to the kilo code sidebar
- Each agent is driven by a in process isolated kilo code cli binary
- Users can start multiple agent sessions in parallel in the same repository
- Start, stop, and remove sessions
- Session state managed via Jotai in webview

**Notes:**
- The parallel worktree integration is not part of this PR to keep it smaller
- Behind feature flag (`AGENT_MANAGER_ENABLED`) - dev mode only
- Agents run in full-auto mode without user confirmation, but user gets a notification
- Requires kilocode CLI to be installed (we validate this)
- We also validate for empty repositories
- All localizations are added

**Screens**
image
image
image
image

### CODE REVIEW

File: packages/types/src/feature-flags.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// kilocode_change - new file
```
Comment (RSO) [inline packages/types/src/feature-flags.ts]: Seeing how `MODEL_SELECTION_ENABLED` doesn't have `// kilocode_change` I guess we're fine without it.

File: src/activate/registerCommands.ts [hunk 2]
```diff
@@ -65,8 +66,22 @@ export type RegisterCommandOptions = {
...
 
+// kilocode_change start - Agent Manager provider instance (singleton for the extension)
+let agentManagerProvider: AgentManagerProvider | undefined
+// kilocode_change end
+
 export const registerCommands = (options: RegisterCommandOptions) => {
-	const { context } = options
+	const { context, outputChannel } = options
+
+	// kilocode_change start - Set feature flag context for Agent Manager visibility
```
Comment (RSO) [inline src/activate/registerCommands.ts]: I guess we should move this one up a few lines?

File: src/core/kilocode/agent-manager/AgentManagerProvider.ts [hunk 3]
```diff
@@ -0,0 +1,623 @@
+import * as vscode from "vscode"
```
Comment (RSO) [inline src/core/kilocode/agent-manager/AgentManagerProvider.ts:1]: Let's move this file to a `kilocode/` folder to make it clear it's Kilo-only

File: src/core/agent-manager/AgentManagerProvider.ts [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+import { AgentRegistry } from "./AgentRegistry"
+import { CliOutputParser, type StreamEvent, type KilocodeStreamEvent, type KilocodePayload } from "./CliOutputParser"
+import { getUri } from "../webview/getUri"
+import { getNonce } from "../webview/getNonce"
+import type { ClineMessage } from "@roo-code/types"
+
+const SESSION_TIMEOUT_MS = 120_000 // 2 minutes
+const WINDOWS = "win32"
+
+export function getKilocodeCliCandidatePaths(env: NodeJS.ProcessEnv, platform: NodeJS.Platform): string[] {
```
Comment (RSO) [inline src/core/agent-manager/AgentManagerProvider.ts]: I'm not sure about this implementation. Can't we use the `PATH` to look up `kilocode`? Since I use `mise-en-place` to install NodeJS, my `kilocode` bin is located at `/Users/remon/.local/share/mise/installs/node/22.15.1/bin/kilo`, and from the looks of it I'm not sure if this function will detect that.

File: src/core/kilocode/agent-manager/AgentManagerProvider.ts [hunk 5]
```diff
@@ -0,0 +1,623 @@
...
+		for (const path of this.cliCandidatePaths) {
+			try {
+				if (path && (await fileExistsAtPath(path))) {
+					this.cliPath = path
+					return path
+				}
+			} catch (error) {
+				this.outputChannel.appendLine(`[AgentManager] Error checking CLI path ${path}: ${error}`)
+			}
+		}
```
Comment (RSO) [inline src/core/kilocode/agent-manager/AgentManagerProvider.ts]: I don't know about this whole thing. It feels overly complex, and like we're preparing for a use case that we don't understand. If `kilocode` is not on the path, I just think we should throw an error and have a docs page or something that explains to users that we expect `kilo` to be on the path.

Comment (RSO) [reply src/core/kilocode/agent-manager/AgentManagerProvider.ts]: My worry about logic like this is that it takes up a lot of space in files and is often hard to understand. And since nobody should run through this (because they should have `kilocode` on their PATH), when they do, we don't know that the outcome of the method will be the outcome that we expect.

File: src/core/agent-manager/AgentRegistry.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+	public setSelectedId(id: string | null): void {
+		this.selectedId = id
+	}
+
+	/**
+	 * Get the selected session ID
+	 */
+	public getSelectedId(): string | null {
+		return this.selectedId
+	}
```
Comment (RSO) [inline src/core/agent-manager/AgentRegistry.ts]: Does JS not have getters/setters that are cleaner than this?

File: src/core/task/Task.ts [hunk 7]
```diff
@@ -416,7 +417,8 @@ export class Task extends EventEmitter<TaskEvents> implements TaskLike {
 		this.fuzzyMatchThreshold = fuzzyMatchThreshold
 		this.consecutiveMistakeLimit = consecutiveMistakeLimit ?? DEFAULT_CONSECUTIVE_MISTAKE_LIMIT
 		this.providerRef = new WeakRef(provider)
-		this.globalStoragePath = provider.context.globalStorageUri.fsPath
+		// Handle CLI mode where globalStorageUri might not be properly set
+		this.globalStoragePath = provider.context?.globalStorageUri?.fsPath ?? this.getCliGlobalStoragePath()
```
Comment (RSO) [inline src/core/task/Task.ts]:
```suggestion
		// kilocode_change start: Handle CLI mode where globalStorageUri might not be properly set
		this.globalStoragePath = provider.context?.globalStorageUri?.fsPath ?? this.getCliGlobalStoragePath()
		// kilocode_change end
```

**Review by RSO — CHANGES_REQUESTED** [review]
Let's move all the `agent-manager` directories to `kilocode/` subdirectories to prevent potential merge conflicts. Also, I tried to test the Agent Manager, but I got an empty screen:

Image

File: packages/types/src/feature-flags.ts (hunk 1 above)
Comment (marius-kilocode) [reply packages/types/src/feature-flags.ts]: Agreed, make sense.

File: src/activate/registerCommands.ts (hunk 2 above)
Comment (marius-kilocode) [reply src/activate/registerCommands.ts]: Agreed, I went one step further and extracted it, that should reduce conflicts.

File: src/core/kilocode/agent-manager/AgentManagerProvider.ts (hunk 3 above)
Comment (marius-kilocode) [reply src/core/kilocode/agent-manager/AgentManagerProvider.ts:1]: I will move them to the existing kilocode dirs.
I noticed that the existing core logic lives in `src/core/kilocode/` instead of `src/kilocode/core` was there an intentional decision to cut it in core @RSO ?

Comment (marius-kilocode) [reply src/core/kilocode/agent-manager/AgentManagerProvider.ts:1]: Regarding the empty screen, I didn't add hot module replacement support like the main ClineProvider has. This required you to run `pnpm build` before testing it in VsCode manually. I will add that as well. So it's easier to test in development.

File: src/core/agent-manager/AgentRegistry.ts (hunk 6 above)
Comment (marius-kilocode) [reply src/core/agent-manager/AgentRegistry.ts]: PlainOldJavaScript :) , the component wan't the cleanest. I replaced setter/getter, magic constants and wrote some tests. Anyway, this will need to be rewritten once we integrate session persistence.

File: src/core/kilocode/agent-manager/AgentManagerProvider.ts (hunk 5 above)
Comment (marius-kilocode) [reply src/core/kilocode/agent-manager/AgentManagerProvider.ts]: image
We actually already handle the edge case when the CLI is not installed, I guess this was the whole point of the detection. But it's a bit over-engineered, I simplified it significantly and extracted it so it should be easy to reuse and adapt later on.

**Review by RSO — APPROVED** [review]
I tested it, and while there's definitely room for improvement, I think this is a good place to start with a first version. Great job!
